### PR TITLE
Extensions outside of document root for composer-based TYPO3 installations

### PR DESCRIPTION
**Pull request by linawolf**
references https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/203

### CODE REVIEW

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 1]
```diff
@@ -1,127 +1,205 @@
...
+Directory :file:`typo3temp/assets/` contains temporary files that should be
+public available. This includes generated images and compressed css and
+JavaScript files.
+
+.. _directory-var:
+
+:file:`var/`
+============
+
+Directory for temporary files that contains private files (e.g. cached
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:187]:
```suggestion
Directory for temporary files that contains private files (e.g.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 2]
```diff
@@ -1,127 +1,205 @@
...
+Fluid templates) and should not be publicly available.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:188]:
```suggestion
cache and logs files) and should not be publicly available.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 3]
```diff
@@ -1,127 +1,205 @@
...
+See also :ref:`Environment-configuring-paths` for a more detailed description.
+
+.. _directory-var-labels:
+
+:file:`labels/`
+---------------
+
+The directory :file:`var/labels/` for extension localisations. Contains all
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:196]:
```suggestion
The directory :file:`var/labels/` is for extension localisations. It contains all
```

We should use full sentences.

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstalations.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
+..  include:: /Includes.rst.txt
+..  index::
+    Path; Legacy installations
+    see: Directory structure; Path
+..  _legacy-directory-structure:
+
+=========================================
+Legacy installations: Directory structure
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstalations.rst:8]: The file name should be adjusted to "LegacyInstallation" (double L).

Also, I suggest to not add a separate file for legacy, but integrate it on the index page. Maybe the differences just beyond the Composer-based folders.

This way we don't have to maintain two files and users looking for the structure via search engine get on main result, not two. I think, two files may confuse users which just scan the page.

So, I haven't reviewed this page for now.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 5]
```diff
@@ -34,9 +32,10 @@ Environment PHP API
 getProjectPath()
 ----------------
 
-The environment provides the path to the folder containing the :file:`composer.json`.
-For projects without Composer setup, this is equal to :ref:`Environment-public-path`.
+The environment provides the path to the `top most directory <directory-project>`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:35]:
```suggestion
The method provides the path to the `top most directory <directory-project>`
```

Also: Why is "top most directory <directory-project>" highlighted?

File: Documentation/ApiOverview/Environment/Index.rst [hunk 6]
```diff
@@ -34,9 +32,10 @@ Environment PHP API
...
+containing the :file:`composer.json`.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:36]:
```suggestion
containing the project's :file:`composer.json`.
```

Just to make it clear.

But: legacy installations don't have a project's composer.json.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 7]
```diff
@@ -46,10 +45,12 @@ For projects without Composer setup, this is equal to :ref:`Environment-public-p
 getPublicPath()
 ---------------
 
-The environment provides the path to the public web folder with
-:file:`index.php` for the TYPO3 frontend. This was previously :php:`PATH_site`.
-For projects without Composer setup, this is equal to :ref:`Environment-project-path`.
+The environment provides the path to the :ref:`public web folder <directory-public>`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:48]:
```suggestion
The method provides the path to the :ref:`public web folder <directory-public>`
```

File: Documentation/ApiOverview/Environment/Index.rst [hunk 8]
```diff
@@ -60,16 +61,16 @@ For projects without Composer setup, this is equal to :ref:`Environment-project-
 getVarPath()
 ------------
 
-The environment provides the path to the :file:`var` folder. This folder contains
-data like logs, sessions, locks, and cache files.
+The environment provides the path to the :ref:`var <directory>` folder.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:64]:
```suggestion
The method provides the path to the :ref:`var <directory>` folder.
```

File: Documentation/ApiOverview/Environment/Index.rst [hunk 9]
```diff
@@ -97,14 +98,14 @@ the web document root - a situation that is not optimal from a security point of
 getLabelsPath()
 ---------------
 
-The environment provides the path to :file:`labels`, respective :file:`l10n`
+The environment provides the path to :ref:`var/labels/ <directory-var-labels>`,
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:101]:
```suggestion
The method provides the path to :ref:`var/labels/ <directory-var-labels>`,
```

File: Documentation/Configuration/ConfigurationFiles.rst [hunk 10]
```diff
@@ -0,0 +1,26 @@
+.. include:: /Includes.rst.txt
+
+.. _configuration-files:
+
+===================
+Configuration files
```
Comment (brotkrueml) [inline Documentation/Configuration/ConfigurationFiles.rst:6]: I suggest to do this change in a separate PR, so it can be merged also to 11.5. This has nothing to do with cms-composer-installers v5.

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 11]
```diff
@@ -1,55 +1,59 @@
...
 ===============================
 
-The files for an extension are located in a folder named by the
-*extension key*. The location of this folder can be either inside
-`typo3/sysext/` or `typo3conf/ext/`.
-
-The extension *must* be programmed so that it does automatically
-detect where it is located and can work from both locations.
-
+The files for an extension are located in a folder named vendor, followed by
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:9]: This is not really true. For an integrator or developer the extensions are placed in a "packages" or "local_packages" folder, which then gets symlinked via Composer into vendor.

This sounds now, that I have to place my extensions directly under the "vendor" folder.

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 12]
```diff
@@ -1,55 +1,59 @@
...
+a folder with the composer name. The location of this folder is in the directory
+:ref:`vendor/ <directory-vendor>`.
 
+In legacy installations they are found in `typo3/sysext/` or `typo3conf/ext/`.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:13]:
```suggestion
In legacy installations they are found in `typo3/sysext/` (system extensions) or `typo3conf/ext/` (local extensions).
```

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 13]
```diff
@@ -1,55 +1,59 @@
...
 
 .. _extension-local:
 
 Local Extensions
 ================
 
-Local extensions are located in the :file:`typo3conf/ext/` directory.
+Local extensions have the composer type `typo3-cms-extension`:
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:20]:
```suggestion
Local extensions have the Composer type `typo3-cms-extension`:
```

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 14]
```diff
@@ -1,55 +1,59 @@
...
+..  code-block:: json
+    :caption: EXT:my_extension/composer.json`
 
+    {
+        "name": "myvendor/my-extension",
+        "type": "typo3-cms-extension",
+        "...": "..."
+    }
+
+The extension will be installed in the directory :ref:`vendor <directory-vendor>`
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:31]: The sentence sounds bumpy. Also: We should differentiate between stored (by the developer) and installed (by Composer). It is not obvious what is meant here.

Also: "vendor" is not linked.

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst [hunk 15]
```diff
@@ -1,55 +1,59 @@
...
-.. _extension-loading-precedence:
+    {
+        "name": "typo3/cms-core",
+        "type": "typo3-cms-framework",
+        "...": "..."
+    }
 
-Loading Precedence
-==================
+The extensions will be installed in the directory :ref:`vendor <directory-vendor>`.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:55]: "vendor" is not linked.

File: Documentation/ExtensionArchitecture/FileStructure/Configuration/Index.rst [hunk 16]
```diff
@@ -23,7 +23,7 @@ folder:
 .. code-block:: none
    :caption: Common content of the configuration folder
 
-   $ tree public/typo3conf/ext/my_extension/Configuration/TypoScript/
+   $ tree vendor/myvendor/my-extension/Configuration/
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/FileStructure/Configuration/Index.rst:26]: We should stick to "packages/local_packages". We mix two things up: storing (by a developer) and installed (by Composer). Also examples below.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 17]
```diff
@@ -1,127 +1,205 @@
...
+
+The folder :file:`config/sites` contains subfolders for each
+:ref:`site configuration <sitehandling>`.
+
+.. _directory-local_packages:
+
+:file:`local_packages` (optional)
+=================================
+
+Each web site which is run on TYPO3 **should** have a sitepackage, a special
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:49]: Its purpose is a special one. It does what a theme would do in other CMS. So it is confusing for newcommers that everything is an extension in TYPO3.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 18]
```diff
@@ -1,127 +1,205 @@
...
+
+:file:`sites/`
+--------------
+
+The folder :file:`config/sites` contains subfolders for each
+:ref:`site configuration <sitehandling>`.
+
+.. _directory-local_packages:
+
+:file:`local_packages` (optional)
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:46]: Here we use "local_packages": https://docs.typo3.org/m/typo3/tutorial-sitepackage/main/en-us/ExtensionInstallation/Index.html

I kind of prefere this name because it makes clear that it is a local package not one installed from somewhere else, but I agree, we should call it the same all throughout the documentation

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 19]
```diff
@@ -1,127 +1,205 @@
...
+:file:`_assets/`.
+
+..  note::
+    In TYPO3 v12 using the `typo3/cms-composer-installers` in version
+    5 is mandatory. Therefore the publicly available files provided by
+    extensions are now always stored in this directory.
+
+.. _directory-public-fileadmin:
+
+:file:`fileadmin/`
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:118]: Never done that before, do we have a chapter where we could link for that?

File: Documentation/Configuration/ConfigurationFiles.rst (hunk 10 above)
Comment (linawolf) [reply Documentation/Configuration/ConfigurationFiles.rst:6]: I only moved the information which was in hidden in  the directory structure before. So that I can link from both  Compsoser and Legacy installatins to it

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst (hunk 15 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:55]: will be in my next commit

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 20]
```diff
@@ -1,127 +1,205 @@
...
-      (e.g. generated images).
-
-  - :Directory: :file:`typo3temp/var/`
-    :Description:
-      Directory for temporary files that contains private files (e.g. cached
-      Fluid templates) and should not be publicly available.
-      See also :ref:`Environment-configuring-paths` for a more detailed description.
+.. _directory-project:
+
+Files on project level
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:20]: The new structure has a number of pros: We can link to headlines, not to table cells. We can produce hierarchical content-menüs (added) . We don't waste so much space on the left, expecially on small screens.

I added the complete paths as you suggested.

In my eyes they are sorted alphabetically (by main directory) This becomes also more clear now on using the complete paths.

I don't really like how the `:file` directive looks in headlines and content-menus, I will make a PR in our theme about that.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst (hunk 17 above)
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:49]: rephrased to make this more clear

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstalations.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/LegacyInstalations.rst:8]: There are really no two directories with the same position as there is no public folder. I also kept it short to the minimum and then linked to other chapters.  It would only get more confusing if we only mention some chapters here.

File: Documentation/Configuration/ConfigurationFiles.rst (hunk 10 above)
Comment (linawolf) [reply Documentation/Configuration/ConfigurationFiles.rst:6]: I will make a follow up for v11 then

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst (hunk 11 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst]: Rphrased it

File: Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst (hunk 14 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/Concepts/SystemAndLocalExtensions.rst:31]: Rephrased it

**Comment by linawolf** [comment]
@brotkrueml please have another look

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 21]
```diff
@@ -203,3 +203,29 @@ This path can be retrieved from the Environment API, see :ref:`Environment-label
...
     LegacyInstalations
+
+
+.. _directory-vendor:
+
+:file:`vendor/`
+===============
+
+In this directory, which lies outside of the webroot, all extensions (system,
+third party and local) are installed as composer packages.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:221]:
```suggestion
third-party and local) are installed as Composer packages.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 22]
```diff
@@ -203,3 +203,29 @@ This path can be retrieved from the Environment API, see :ref:`Environment-label
...
+
+The directory contains folders for each required vendor and inside each
+vendor directory there is a folder for the composer name.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:224]:
```suggestion
vendor directory there is a folder with the different project names.
```
There is no "composer name".

See: https://getcomposer.org/doc/01-basic-usage.md#package-names

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 23]
```diff
@@ -203,3 +203,29 @@ This path can be retrieved from the Environment API, see :ref:`Environment-label
...
+
+For example the system extension `core` has the complete composer name
+`typo3/cms-core` and will therefore be installed into the directory
+:file:`vendor/typo3/cms-core`. The extension `news`, composer name
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:228]:
```suggestion
:file:`vendor/typo3/cms-core`. The extension `news`, package name
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 24]
```diff
@@ -203,3 +203,29 @@ This path can be retrieved from the Environment API, see :ref:`Environment-label
...
+`georgringer/news` will be installed into the folder
+:file:`vendor/georgringer/news`.
+
+Never put or symlink your extensions manually into this directory as they would
+not be found. Local extensions and sitepackages should be kept in a separate
+folder outside the web root, for example
+:ref:`local_packages <directory-local_packages>`. This directory then be
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:235]: This sentence sounds bumpy.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 25]
```diff
@@ -1,127 +1,244 @@
...
-main directories within the web server document root. You will find
-this structure to be almost always like that. Depending on the installation
-variant you choose however, this may be slightly different. For instance,
-it is possible to have all PHP files except the entry points :file:`index.php`
-within the Composer-managed :file:`vendor/` directory, outside of the document
-root. This setup however did not fully settle yet, and is not documented
-here in detail. So, if you look at "casual" TYPO3 installations, you will
-almost always find the directory structure as outlined below.
+The structure below describes the directory structure in a typical
+Composer-base TYPO3 installation. For the structure in a legacy installation
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:12]:
```suggestion
Composer-based TYPO3 installation. For the structure in a legacy installation
```

Also, that this page is for Composer-based installations should be added to the heading: "Directory structure (Composer-based installations)" or something to make it clear. Not everyone reads the first sentence when he/she/it looks for a path, but the heading is more prominent.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 26]
```diff
@@ -1,127 +1,244 @@
...
+The folder :file:`config/sites` contains subfolders for each
+:ref:`site configuration <sitehandling>`.
+
+.. _directory-local_packages:
+
+:file:`local_packages/`
+-----------------------
+
+Each web site which is run on TYPO3 **should** have a sitepackage, an
+extension with a special purpose containing all templates, styles, images
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:57]:
```suggestion
extension with a special purpose containing all templates, styles, images,
```

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+:ref:`Configuration files <configuration-files>` for details.
+
+This path can be retrieved from the Environment API, see :ref:`Environment-config-path`.
+
+.. _legacy-directory-typo3conf-ext:
+
+:file:`typo3conf/ext/`
+~~~~~~~~~~~~~~~~~~~~~~
+
+Directory for local TYPO3 extensions. Each subdirectory contains one extension.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:78]: What is a "local" TYPO3 extension? Are there also "remote" extensions?

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+
+In Composer-based installations the extensions are installed
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:80]: Why mention Composer?

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst [hunk 29]
```diff
@@ -0,0 +1,126 @@
...
+
+.. _legacy-directory-typo3temp:
+
+:file:`typo3temp/`
+------------------
+
+Directory for temporary files. It contains subdirectories (see below)
+for temporary files of extensions and TYPO3 components.
+
+In Composer-based installations public temorary files can be found in
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:105]: Why mention Composer?

Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:105]:
```suggestion
In Composer-based installations public temporary files can be found in
```

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst [hunk 30]
```diff
@@ -0,0 +1,126 @@
...
+in the directory :ref:`public/typo3temp/ <directory-public-typo3temp>`.
+
+.. _legacy-directory-typo3temp-assets:
+
+:file:`typo3temp/assets/`
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Directory for temporary files that should be public available
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:113]:
```suggestion
Directory for temporary files that should be publicly available
```

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst [hunk 31]
```diff
@@ -0,0 +1,126 @@
...
+.. _legacy-directory-typo3temp-var:
+
+:file:`typo3temp/var/`
+~~~~~~~~~~~~~~~~~~~~~~
+
+Directory for temporary files that contains private files (e.g. cached
+Fluid templates) and should not be publicly available.
+See also :ref:`Environment-configuring-paths` for a more detailed description.
+
+In Composer-based installations private temporary files can be found in
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:125]: Why mention Composer?

File: Documentation/ApiOverview/Environment/Index.rst [hunk 32]
```diff
@@ -34,9 +32,10 @@ Environment PHP API
 getProjectPath()
 ----------------
 
-The environment provides the path to the folder containing the :file:`composer.json`.
-For projects without Composer setup, this is equal to :ref:`Environment-public-path`.
+The method provides the path to the `top most directory <directory-project>`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:35]:
```suggestion
The method provides the path to the :ref:`top most directory <directory-project>`
```

File: Documentation/ApiOverview/Environment/Index.rst [hunk 33]
```diff
@@ -34,9 +32,10 @@ Environment PHP API
...
+containing the project's :file:`composer.json`.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:36]:
```suggestion
containing the project's :file:`composer.json` file.
```

File: Documentation/ApiOverview/Environment/Index.rst [hunk 34]
```diff
@@ -46,10 +45,12 @@ For projects without Composer setup, this is equal to :ref:`Environment-public-p
 getPublicPath()
 ---------------
 
-The environment provides the path to the public web folder with
-:file:`index.php` for the TYPO3 frontend. This was previously :php:`PATH_site`.
-For projects without Composer setup, this is equal to :ref:`Environment-project-path`.
+The method provides the path to the :ref:`public web folder <directory-public>`
+with :file:`index.php` for the TYPO3 frontend.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:49]: What is a TYPO3 frontend? Yes I know, it was already there and can be part of another PR.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 35]
```diff
@@ -1,127 +1,244 @@
...
+:file:`vendor/typo3/cms-core`. The extension `news`, composer name
+`georgringer/news` will be installed into the folder
+:file:`vendor/georgringer/news`.
+
+Never put or symlink your extensions manually into this directory as they would
+not be found. Local extensions and sitepackages should be kept in a separate
+folder outside the web root, for example
+:ref:`local_packages <directory-local_packages>`. This directory then be
+defined as path in the repository section of the :file:`project/composer.json`.
+Upon installing it, it will be symlinked into the folder
```
Comment (linawolf) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:237]:
```suggestion
Upon installation , Composer creates a symlink to 
```

File: Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst [hunk 36]
```diff
@@ -0,0 +1,126 @@
...
+~~~~~~~~~~~~~~~~~~~~~
+
+All system extensions, supplied by the TYPO3 Core, are stored here.  For example,
+the `frontend` extension amongst other things contains the
+"TypoScript library", the code for generating the Frontend website. In
+each system extension the PHP files are located in the folder
+:file:`Classes/`. See :ref:`extension files locations <extension-files-locations>`
+for more information on how single extensions are structured.
+
+In Composer-based installations the entry PHP files can be found in the directory
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/LegacyInstallations.rst:58]: To facilitate migration to the Composer-based structure I thought?

File: Documentation/ApiOverview/Environment/Index.rst (hunk 34 above)
Comment (linawolf) [reply Documentation/ApiOverview/Environment/Index.rst:49]: I think we are mentioning Frontend and Backend all over TYPO3 explained...

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 37]
```diff
@@ -1,127 +1,244 @@
...
+:file:`vendor/myvendor/my-extension` by Composer.
```
Comment (linawolf) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:238]:
```suggestion
:file:`vendor/myvendor/my-extension`.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 38]
```diff
@@ -1,127 +1,244 @@
...
+:file:`vendor/typo3/cms-core`. The extension `news`, package name
+`georgringer/news` will be installed into the folder
+:file:`vendor/georgringer/news`.
+
+Never put or symlink your extensions manually into this directory as they would
+not be found. Local extensions and sitepackages should be kept in a separate
+folder outside the web root, for example
+:ref:`local_packages <directory-local_packages>`. This directory then be
+defined as path in the repository section of the :file:`project/composer.json`.
+Upon installation , Composer creates a symlink to
```
Comment (linawolf) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:237]:
```suggestion
:ref:`local_packages <directory-local_packages>`. 
Upon installation , Composer creates a symlink from local_packages to 
```

**Comment by linawolf** [comment]
@brotkrueml so let us do https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/2241 as a first step?

**Comment by linawolf** [comment]
Will be solved by https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/2279
